### PR TITLE
feat: ability to add phone number in profile 📲

### LINKED ISSUE #471: Add/Edit Phone Numbers in Member Profile 🔢
### Description

We want to implement the ability to add/edit phone numbers from the Member Profile. Specifically, this should appear as a field under `/profile/general`. 

### Acceptance Criteria

To implement this issue, the following is required:
- add a field `phone_number` to the database
- add a new field to `/profile/general` after the field `CurrentLocationField` [here](https://github.com/colorstackorg/oyster/blob/4b27f72c1841f32b454c2d3d0a1e5149de4b818b/apps/member-profile/app/routes/_profile.profile.general.tsx#L175)
- verify that the value entered in the field is a 10-digit `int`

### Additional Context

You can add a new field to the database using a database migration. See the [docs](https://github.com/colorstackorg/oyster/blob/main/docs/how-to-implement-a-database-migration.md)!

Note that this will also require updating the loader and action functions within [this](https://github.com/colorstackorg/oyster/blob/4b27f72c1841f32b454c2d3d0a1e5149de4b818b/apps/member-profile/app/routes/_profile.profile.general.tsx) file to ensure that any previously entered number is properly loaded, and any added/edited phone number is updated.

### PR DESCRIPTION
**Pull request by Hamza-Mos**
## Description ✏️

Closes #471 

- Implement functionality to add/edit phone numbers in general information section of member profile.

Screenshots:

Empty:

image

Invalid:

image

image

Valid:

image

Database:

image

## Type of Change 🐞

- [x] Feature - A non-breaking change which adds functionality.
- [ ] Fix - A non-breaking change which fixes an issue.
- [ ] Refactor - A change that neither fixes a bug nor adds a feature.
- [ ] Documentation - A change only to in-code or markdown documentation.
- [ ] Tests - A change that adds missing unit/integration tests.
- [ ] Chore - A change that is likely none of the above.

## Checklist ✅

- [x] I have done a self-review of my code.
- [x] I have manually tested my code (if applicable).
- [x] I have added/updated any relevant documentation (if applicable).

### CODE REVIEW

File: apps/member-profile/app/routes/_profile.profile.general.tsx [hunk 1]
```diff
@@ -174,6 +176,16 @@ export default function UpdateGeneralInformationSection() {
           longitudeName={keys.currentLocationLongitude}
         />
 
+        <InputField
+          defaultValue={student.phoneNumber || undefined}
+          description="Enter your 10-digit phone number below (no formatting characters e.g. '-' or '(')."
+          error={errors.phoneNumber}
+          label="Phone Number"
+          name={keys.phoneNumber}
+          placeholder="1234567890"
```
Comment (tomas-salgado) [inline apps/member-profile/app/routes/_profile.profile.general.tsx:185]: nit: let's change the placeholder to "5551234567", since this is more recognizable as a sample phone number.

File: apps/member-profile/app/routes/_profile.profile.general.tsx [hunk 2]
```diff
@@ -174,6 +176,16 @@ export default function UpdateGeneralInformationSection() {
...
+          type="number"
+        />
+
```
Comment (tomas-salgado) [inline apps/member-profile/app/routes/_profile.profile.general.tsx:188]: It might be worth improving the UI here so that we automatically have the formatting (555)-123-4567 rather than just the plain numbers. This can be a separate issue though.

File: packages/ui/src/components/form.tsx [hunk 3]
```diff
@@ -71,7 +71,7 @@ Form.Field = function FormField({
 
 type InputFieldProps = FieldProps<string> &
   Pick<FormFieldProps, 'description' | 'label' | 'required'> &
-  Pick<InputProps, 'disabled' | 'placeholder'>;
+  Pick<InputProps, 'disabled' | 'placeholder' | 'type'>;
```
Comment (tomas-salgado) [inline packages/ui/src/components/form.tsx:74]: I don't think we need to add the changes here for 'type'. We're already verifying that only numbers can be inputted in `student.ts`, and doing this adds an unnecessary icon on the right side of the input field. Lmk if there was any other reason you think we need this 

Screenshot 2024-09-01 at 11 09 25 AM

**Review by tomas-salgado — CHANGES_REQUESTED** [review]
This is great stuff @Hamza-Mos! Just left a few minor comments but almost good to go!

File: apps/member-profile/app/routes/_profile.profile.general.tsx (hunk 2 above)
Comment (Hamza-Mos) [reply apps/member-profile/app/routes/_profile.profile.general.tsx:188]: hmm i can probably tackle this in a separate PR to keep this one small since there are some edge cases with this that I would want to test thoroughly.

Comment (tomas-salgado) [reply apps/member-profile/app/routes/_profile.profile.general.tsx:187]: @Hamza-Mos yeah that looks like what I was thinking. We could go with that and see how it looks

Comment (Hamza-Mos) [reply apps/member-profile/app/routes/_profile.profile.general.tsx:187]: updated @tomas-salgado

**Review by tomas-salgado — APPROVED** [review]
Nice work @Hamza-Mos, LGTM!

**Review by ramiAbdou — APPROVED** [review]
Awesome work, @Hamza-Mos! 💯

Phone number formatting can get very difficult and I thought you killed that `formatPhoneNumber` function! That was a neat trick you used to keep the end parenthesis open when the digit length was 3, which made it easy to delete the 4th number (and similarly for the `-`).

I made an update so that we're only storing digits in our database as opposed to the parentheses/dashes! Thought about storing the country code as well, but don't think we need to since we're only serving US/Canada-based folks. Luckily we avoided a whole slew of complexity by not serving international numbers cause then we would've had to use something like [react-phone-number-input](https://www.npmjs.com/package/react-phone-number-input) which is really heavy and nasty haha.

Can't wait to see this in prod!

File: apps/member-profile/app/routes/_profile.profile.general.tsx (hunk 2 above)
Comment (ramiAbdou) [reply apps/member-profile/app/routes/_profile.profile.general.tsx:193]: The `tel` type doesn't actually do too much in practice, it doesn't automatically format nor validate the input because of the complexity of phone numbers internationally. That being said, you did all the legwork with the `formatPhoneNumber`, that was the key to this PR!
